### PR TITLE
[feat]: [PIE-6637]: Fixing pipeline clone and resource restraint handling for when planExecution doesn't exist

### PR DESCRIPTION
**Pull request by archit-harness**
Fixing pipeline clone and resource restraint handling for when planExecution doesn't exist (happen for queuing of requests)

Impact - Pipeine clone flow and Resource Restratint only If planExecution Doesnt exist.

Testing - Manual running pipeline clone is working

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/40670)

### CODE REVIEW

File: pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/resourcerestraint/service/ResourceRestraintInstanceServiceImpl.java [hunk 1]
```diff
@@ -110,7 +111,14 @@ public boolean updateActiveConstraintsForInstance(ResourceRestraintInstance inst
       switch (scope) {
         case PLAN:
         case PIPELINE:
-          finished = StatusUtils.finalStatuses().contains(planExecutionService.getStatus(releaseEntityId));
+          Status status;
+          try {
+            status = planExecutionService.getStatus(releaseEntityId);
+            finished = StatusUtils.finalStatuses().contains(status);
+          } catch (Exception e) {
+            log.warn("Plan Execution doesn't for the releaseEntityId - " + releaseEntityId);
```
Comment (naidu-harness) [inline pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/resourcerestraint/service/ResourceRestraintInstanceServiceImpl.java:119]: does nt exist?

Comment (archit-harness) [reply pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/resourcerestraint/service/ResourceRestraintInstanceServiceImpl.java:119]: this is coming for execution which are getting queued and then interrupted using paused / abort.
Someone needs to take proper look

Comment (naidu-harness) [reply pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/resourcerestraint/service/ResourceRestraintInstanceServiceImpl.java:119]: ok

**Review by naidu-harness — APPROVED** [review]
:lgtm:

> Reviewed 5 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40670#-:-NHj5wcv70QL4AuYFTWH:bnfp4nl)* status: 5 file(s) to review. 1 of 2 approvals obtained.

**Pull request merged**